packages/esm-patient-biometrics-app/src/biometrics/pagination-see-all.component.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FunctionComponent } from 'react';
+import Pagination from 'carbon-components-react/es/components/Pagination';
+import styles from '@openmrs/esm-patient-common-lib/src/pagination/pagination.component.scss';
+import { ConfigurableLink } from '@openmrs/esm-framework';
+import { usePaginationInfo } from '@openmrs/esm-patient-common-lib';
+
+interface PatientChartPaginationProps {
+  currentItems: number;
+  totalItems: number;
+  pageNumber: number;
+  pageSize: number;
+  onPageNumberChange?: any;
+  pageUrl?: string;
+  urlLabel: string;
+}
+
+export const PatientChartPagination: FunctionComponent<PatientChartPaginationProps> = ({
+  totalItems,
+  pageSize,
+  onPageNumberChange,
+  pageNumber,
+  pageUrl = '',
+  currentItems,
+  urlLabel,
+}) => {
+  const { itemsDisplayed, pageSizes } = usePaginationInfo(pageSize, totalItems, pageNumber, currentItems);
+
+  return (
+    <>
+      {totalItems > 0 && (
+        <div className={styles.paginationContainer}>
+          <div className={styles.paginationLink}>
+            {itemsDisplayed}
+            {pageUrl && (
+              <ConfigurableLink to={pageUrl} className={styles.configurableLink}>
+                {urlLabel}
+              </ConfigurableLink>
+            )}
+          </div>
+          <Pagination
+            className={styles.pagination}
+            page={pageNumber}
+            pageSize={pageSize}
+            pageSizes={pageSizes}
+            totalItems={totalItems}
+            onChange={onPageNumberChange}
+          />
+        </div>
+      )}
```

Comment:
Only commenting once, but there are also other files with the same structure:
In the common lib, there exists the `PatientChartPagination` component which has the exact same structure. Can you reuse that one instead of the duplicate code here?

Reply:
True. I could not use that component from common lib as it had a hard coded attribute (on Ln. 36). This file that I have created is the exact copy but without the hard coded attribute. I did not want to change the original common lib component as it could be getting used else where.

Reply:
Feel free to change that! 😉 If something gets messed up, the build pipeline should catch it before it can cause any issues.

If it's only about the hardcoded "See all", you can just do the following (which is very very unlikely to break something):

```ts
import React from 'react';
import Pagination from 'carbon-components-react/es/components/Pagination';
import styles from './pagination.component.scss';
import { useTranslation } from 'react-i18next';
import { ConfigurableLink } from '@openmrs/esm-framework';
import { usePaginationInfo } from './usePaginationInfo';

interface PatientChartPaginationProps {
  currentItems: number;
  totalItems: number;
  pageNumber: number;
  pageSize: number;
  onPageNumberChange?: any;
  pageUrl?: string;
  linkText?: string; // <- Changed. Make sure to make this prop optional.
}

export const PatientChartPagination: React.FC<PatientChartPaginationProps> = ({
  totalItems,
  pageSize,
  onPageNumberChange,
  pageNumber,
  pageUrl = '',
  currentItems,
  linkText, // <- Changed. 
}) => {
  const { t } = useTranslation();
  const { itemsDisplayed, pageSizes } = usePaginationInfo(pageSize, totalItems, pageNumber, currentItems);

  return (
    <>
      {totalItems > 0 && (
        <div className={styles.paginationContainer}>
          <div className={styles.paginationLink}>
            {itemsDisplayed}
            {pageUrl && (
              <ConfigurableLink to={pageUrl} className={styles.configurableLink}>
                {linkText ?? t('seeAll', 'See all')} // <- Changed. Have a fallback for current users who don't need a special link text.
              </ConfigurableLink>
            )}
          </div>
          <Pagination
            className={styles.pagination}
            page={pageNumber}
            pageSize={pageSize}
            pageSizes={pageSizes}
            totalItems={totalItems}
            onChange={onPageNumberChange}
          />
        </div>
      )}
    </>
  );
};
```

Reply:
This has been implemented. Thanks.